pkg-manager/resolve-dependencies/src/resolvePeers.ts

```diff
@@ -289,16 +289,16 @@ function resolvePeersOfNode<T extends PartialResolvedPackage> (
   const children = node.children
   const parentPkgs = isEmpty(children)
     ? parentParentPkgs
-    : {
-      ...parentParentPkgs,
-      ...toPkgByName(
+    : Object.assign(
+      Object.create(parentParentPkgs),
```

Comment:
This way fields in parentParentPkgs could be replaced.

Reply:
actually, looks like it doesn't override the fields of `parentParentPkgs`

I'll be honest, looks like my knowledge of javascript is not complete. I always used either `Object.assign({}, ...)` or the spread operator and never hear of `Object.create`